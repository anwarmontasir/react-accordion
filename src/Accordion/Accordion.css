.accordion button {
    background: lightblue;
    width: 100%;
    font-size: 20px;
    padding: 5px 0;
}
.accordion p {
    background: #fdfcfc;
    min-height: 200px;
    padding: 20px;
    border: 1px solid #777;
}