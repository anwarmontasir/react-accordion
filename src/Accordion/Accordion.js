import React from 'react';
import './Accordion.css';

class Accordion extends React.Component {
    static defaultProps = { sections: [] };

    state = {
        currentSectionIndex: 0
    }

    handleSectionClick = (index) => {
        this.setState({ currentSectionIndex: index })
    }

    renderSections() {
        const currentSection = this.props.sections[this.state.currentSectionIndex];
        return this.props.sections.map((section, index) => (
           <li>
                <button key={index} onClick={() => this.handleSectionClick(index)}>
                    {section.title}
                </button>
                {this.state.currentSectionIndex === index && <p>{currentSection.content}</p>}
            </li>
        ));
    }

    render() {
        return (
            <ul className="accordion">
                {this.renderSections()}
            </ul>
        )
    }
}

export default Accordion;