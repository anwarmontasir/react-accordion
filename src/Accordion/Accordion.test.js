import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Accordion from './Accordion';

describe('Accordion Component', () => {
    const sectionProp = [
        {
          title: 'Section 1',
          content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
        },
        {
          title: 'Section 2',
          content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
        },
        {
          title: 'Section 3',
          content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
        },
    ];
    it('renders without errors', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Accordion />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders empty given no accordion', () => {
        const wrapper = shallow(<Accordion />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('renders first section by default', () => {
        const wrapper = shallow(<Accordion section={sectionProp} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})