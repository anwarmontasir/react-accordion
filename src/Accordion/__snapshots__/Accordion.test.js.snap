// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component renders empty given no accordion 1`] = `
<ul
  className="accordion"
/>
`;

exports[`Accordion Component renders first section by default 1`] = `
<ul
  className="accordion"
/>
`;
